flutter/news/lib/features/frontpage/data/datasource/articles_remote_data_source.dart

```diff
@@ -1,31 +1,23 @@
-import 'dart:convert';
-
-import 'package:http/http.dart' as http;
-import 'package:news/core/constants.dart';
 import 'package:news/core/error/failures.dart';
+import 'package:news/core/news_api_client.dart';
 import 'package:news/core/result.dart';
-import 'package:news/features/frontpage/data/models/article_model.dart';
-import 'package:news/features/frontpage/data/models/base_news_response_model.dart';
+import 'package:news/features/frontpage/domain/entities/article.dart';
 
 class ArticlesRemoteDataSource {
-  final http.Client client;
-
+  final NewsApiClient client;
   ArticlesRemoteDataSource({required this.client});
 
-  Future<Result<List<ArticleModel>>> topHeadLines() async {
-    var url = Uri.https(Constants.baseUrl, Constants.topHeadlinesEndpoint, {'country': 'us'});
+  Future<Result<List<Article>>> topHeadLines() async {
+    var result = await client
+        .topHeadLines()
+        .then((value) => Result<List<Article>>.success(value.articles))
+        .catchError(
+            (error) => Result<List<Article>>.failure(ServerFailure("TEST")));
```

Comment:
The error handling can be polished in the future in case it's required, atm anything not a `200` is categorized as the same error